File: backend/app/models/bookings.py
from ..extensions import db
from sqlalchemy import Enum
from datetime import datetime


class Booking(db.Model):
    __tablename__ = "bookings"
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
    user = db.relationship("User", backref="bookings")
    event_id = db.Column(db.Integer, db.ForeignKey("events.id"), nullable=False)
    event = db.relationship("Event", backref="bookings")
    booking_status = db.Column(
        Enum("pending", "confirmed", "canceled", name="booking_status"),
        default="pending",
        nullable=False,
    )
    created_at = db.Column(db.DateTime, default=datetime.now(), nullable=False)
    updated_at = db.Column(
        db.DateTime, default=datetime.now(), onupdate=datetime.now(), nullable=False
    )

File: backend/internal/models/bookings.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type BookingStatus string

const (
	BookingPending   BookingStatus = "pending"
	BookingConfirmed BookingStatus = "confirmed"
	BookingCanceled  BookingStatus = "canceled"
)

type Booking struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	UserID    uint           `gorm:"not null" json:"user_id"`
	User      *User          `gorm:"foreignKey:UserID" json:"user,omitempty"`
	EventID   uint           `gorm:"not null" json:"event_id"`
	Event     *Event         `gorm:"foreignKey:EventID" json:"event,omitempty"`
	Status    BookingStatus  `gorm:"type:text;check:status IN ('pending', 'confirmed', 'canceled');default:'pending';not null" json:"status"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}
